NitroxClient/Communication/Packets/Processors/RemoveCreatureCorpseProcessor.cs

```diff
@@ -34,38 +33,42 @@ public override void Process(RemoveCreatureCorpse packet)
 
         creatureDeath.transform.localPosition = packet.DeathPosition.ToUnity();
         creatureDeath.transform.localRotation = packet.DeathRotation.ToUnity();
-        CoroutineHost.StartCoroutine(SimplerOnKillAsync(creatureDeath, packet.CreatureId));
+
+        SafeOnKillAsync(creatureDeath, packet.CreatureId, simulationOwnership, liveMixinManager);
     }
 
     /// <summary>
     /// Calls only some parts from <see cref="CreatureDeath.OnKillAsync"/> to avoid sending packets from it
     /// or already synced behaviour (like spawning another respawner from the remote clients)
     /// </summary>
-    public IEnumerator SimplerOnKillAsync(CreatureDeath creatureDeath, NitroxId creatureId)
+    public static void SafeOnKillAsync(CreatureDeath creatureDeath, NitroxId creatureId, SimulationOwnership simulationOwnership, LiveMixinManager liveMixinManager)
     {
         // Ensure we don't broadcast anything from this kill event
         simulationOwnership.StopSimulatingEntity(creatureId);
 
         // Remove the position broadcasting stuff from it
         EntityPositionBroadcaster.RemoveEntityMovementControl(creatureDeath.gameObject, creatureId);
 
-        // Receiving this packet means the creature is dead
-        liveMixinManager.SyncRemoteHealth(creatureDeath.liveMixin, 0);
-
         // To avoid SpawnRespawner to be called
         creatureDeath.respawn = false;
         creatureDeath.hasSpawnedRespawner = true;
 
         // To avoid the cooked data section
-        bool lastDamageWasHeat = creatureDeath.lastDamageWasHeat;
         creatureDeath.lastDamageWasHeat = false;
 
+        // Receiving this packet means the creature is dead
```

Comment:
The goal here is to ensure the fish is 100% dead when turning it into a corpse but without triggering any side-effects like broadcasting packets since we're currently processing one